### ISSUE TITLE
Extract new no-single-line-block-comment rule from comment-wrapping rule

### ISSUE THREAD

**Issue by atulgpt** [pinterest/ktlint#1980]
## Expected Behavior
Comments like `/* no-op */` should not be reported by the rule `CommentWrapping` as in Kotlin these are auto suggested by IDE when typing `noop`

## Observed Behavior
`Ktlint` reports the `/* no-op */` as this is single line comment, ktlint suggests to use `// no-op`

## Your Environment
* Version of ktlint used: Using `Ktlint` by `detekt` at version 0.49.0

**Comment by paul-dingemans** [comment]
I can not reproduce this behavior in Intellij. Typing `noop` does notihing at all. Maybe this behavior is cause by a live template or a plugin. But even if IntelliJ would generate `/* no-op */` then I would still not agree to keep it that way. The only acceptable reason would be if `/* no-op */` would have a different semantic meaning then `// no-op`. For example the `/* ktlint-disable */` comment can not be replace with `// ktlint-disable` as the first one starts disabling for a block of lines while the latter only disables the current line.

**Comment by paul-dingemans** [comment]
Ok, cool. Tnx for clarifying that. But still, I do not think it is necessary to keep the block comment if it can be replaced with a EOL comment.

**Comment by RBusarow** [comment]
@paul-dingemans could this specific maybe be pulled into a different rule, so that it can be toggled individually?

**Comment by paul-dingemans** [comment]
What compelling reason do you have for that? Is Android Studio not working properly anymore after replacing `/* no-op */` with `// no-op`?

**Comment by RBusarow** [comment]
I'm not sure that "does the rule break the IDE?" is the right litmus test for this.

The rule does two things.  There's nothing inherent in "external content wrapping" which says that block comments should be converted to EOL comments.  It simply makes sense to decouple them.

**Comment by paul-dingemans** [comment]
> The rule does two things. There's nothing inherent in "external content wrapping" which says that block comments should be converted to EOL comments. It simply makes sense to decouple them.

True, that is a valid argument.

**Comment by atulgpt** [comment]
Hi, @paul-dingemans is this issue up for grabs?

**Comment by paul-dingemans** [comment]
Yes sure, please give it a try. But, I want to try to release the `0.49.1` within the next week. If that puts to much stress on you, you might want to pick up something else/

**Comment by paul-dingemans** [comment]
@atulgpt 
This is the last outstanding issue for the `0.49.1` bugfix release. I will pick it up myself, so that I can start integration testing of the release. Issue #1938 might be a nice once to get started with this code base.

**Comment by atulgpt** [comment]
Hi @paul-dingemans, sure I will pick https://github.com/pinterest/ktlint/issues/1938

**Comment by atulgpt** [comment]
Hi @paul-dingemans in regards to original issues, I still think `/* no-op */` should be allowed as `// no-op` is not valid for all code snippets. For example below code is not a valid one
```
val foo = { // no-op }
```
so we must use 
```
 val foo = { /* no-op */ }
```
in this case. Although in this case, comment wrapping doesn't report anything. But this can create inconsistency throughout the code

**Comment by paul-dingemans** [comment]
> Hi @paul-dingemans in regards to original issues, I still think `/* no-op */` should be allowed as `// no-op` is not valid for all code snippets. For example below code is not a valid one
> 
> ```
> val foo = { // no-op }
> ```
> 
> so we must use
> 
> ```
>  val foo = { /* no-op */ }
> ```
> 
> in this case. Although in this case, comment wrapping doesn't report anything. But this can create inconsistency throughout the code

This is already catered for.
```
    @Test
    fun `Given a single line block containing a block comment then do not reformat`() {
        val code =
            """
            val foo = { /* no-op */ }
            """.trimIndent()
        noSingleLineBlockCommentRuleAssertThat(code)
            .withEditorConfigOverride(CODE_STYLE_PROPERTY to CodeStyleValue.ktlint_official)
            .hasNoLintViolations()
    }
```
